js-pkg/packages/driftdb-react/src/index.tsx

```diff
@@ -13,59 +13,104 @@ import React, { useCallback, useEffect, useRef, useState } from 'react'
 
 const ROOM_ID_KEY = '_driftdb_room'
 
-export const DatabaseContext = React.createContext<DbConnection | null>(null)
-
-export function useDatabase(): DbConnection {
-  const db = React.useContext(DatabaseContext)
-  if (db === null) {
-    throw new Error('useDatabase must be used within a DriftDBProvider')
+/**
+ * A React component that provides a `DbConnection` to all child components.
+ * 
+ * @param props The props for the component.
+ */
+export function DriftDBProvider(props: {
+  /** Elements under the prodider in the tree. */
+  children: React.ReactNode
+  /** The URL of the DriftDB API. */
+  api: string
+  /** The room ID to connect to. If not provided, attempts to extract the room ID
+   *  from the URL and creates a new room if one is not present. */
+  room?: string
+  /** Whether to use binary messages (enables raw typed arrays in messages). */
+  useBinary?: boolean
```

Comment:
Since the `use` prefix implies hooks these days, I might suggest changing this to just `binary` or `cbor` or something? I think it would help with clarity when talking about the API. Or maybe make this the default and have the developer opt into `json`?

Reply:
Haha, I think we've gone full circle. It was originally `binary` but we were worried about that not being clear, then it was `cbor` but we felt that exposed an implementation detail. I see what you mean about `use` implying a hook though, hmm.

Maybe `binaryProtocol`? Or `allowTypedArrays`?

Reply:
Oh whoops - I was thinking we had landed on `cbor`, but I'm probably forgetting conversations we had after that. I kinda feel like the format _is_ an implementation detail and so the developer shouldn't care what we use under the hood. But if we're saying there's sometimes a need to use JSON for debugging, then I guess we're saying it's _not_ an implementation detail and that developers really _do_ need to know about the message format. I kinda think making json opt-in is the best approach, but I don't want to block you. I'll let you decide!

Reply:
I want to understand the performance implications of `cbor` better before making it the default. I'm inclined to stick to `useBinary`; even though I agree with the `use*` resembling a hook, I think it's clear enough that it's a prop in this context and not a hook.

Reply:
Makes sense!